phpunit/class-block-templates-test.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * REST API: Block_Templates_Test class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Tests for the Block Templates abstraction layer.
+ */
+class Block_Templates_Test extends WP_UnitTestCase {
```

Comment:
REST API controller tests should subclass `WP_Test_REST_Controller_Testcase`. That should fix the routes not appearing, and will also provide a framework for the base line of test coverage.

Reply:
Ah, I see so this also has general PHP API tests. We should end up with two different PHPUnit classes then. One for the general PHP API, and one for the REST API that subclasses the REST API specific base class.

Reply:
I got this issue when trying to create the REST one https://github.com/WordPress/gutenberg/pull/27910#issuecomment-753868286

Reply:
Oh so the endpoint issue might be related to the theme as well.

Reply:
Yeah if the CPT also is registered conditionally you'd need to apply the same fix.